k-distribution/include/kframework/builtin/domains.md

```diff
@@ -1399,8 +1399,14 @@ a single code point) and an integer in O(1) time.
 ### String substring
 
 You can compute a substring of a string in O(N) time (where N is the
-length of the substring). Note that the range generated is 
-`[startIndex..endIndex)`
+length of the substring). There are two important facts to note:
+
+1.  the range generated includes the character at `startIndex` but excludes the
+    character at `endIndex`, i.e., the range is `[startIndex..endIndex)`.
+2.  when using the LLVM backend, if the substring indices are invalid (i.e.,
```

Comment:
I imagine on the Haskell backend, in this case it returns #Bottom, which is not what you want, so you probably just want to say "This function only returns valid results if" rather than singling out a specific backend and explaining that it causes it to crash.

Reply:
#1820 which was the inspiration for this change was about the frustration caused by not knowing that builtin functions may throw exceptions when passed invalid input. If the comment doesn't mention that part, then I don't think it will address the issue very well. I can remove the part about being specific to the LLVM backend though without any harm though (since I just "may throw"). What do you think?

There are other ways to achieve a similar effect that we can try, i.e. add a `may-throw(...)` attribute parameterized by a list of backend names that we can attach to symbols with the `function` attribute. Then we can add a new warning class about functions which throw exceptions.

Reply:
I mean, the point is, if it doesn't have the functional attribute, it's not defined on all inputs. The documentation can describe what inputs it's defined on, and we can explain at the top of the documentation what happens on various backends if you call a function on inputs it's not defined on, but we shouldn't be going into detail on individual backends in this particular place in the documentation, because then we'd have to do it everywhere.

Reply:
@dwightguth I updated the documentation according to your recommendation. Would you mind taking a look at this again?